src/QuadForm/indefiniteLLL.jl

```diff
@@ -0,0 +1,316 @@
+
+################################################################################
+#
+#  Indefinite LLL reduction
+#
+################################################################################
+
+################################################################################
+#  Helpful
+################################################################################
+
+function _vecextract(M::MatElem, x::Union{Int64,Vector{Int64}},y::Int64)
+  y_bin = digits(y, base = 2)
+  #=if length(y_bin) > length(M) || issubset(x,range(1,length(M)))
+    error("Exceeds the size of the matrix")
+  end =#
+  list_y = [i for i = 1:length(y_bin) if y_bin[i] == 1]
+
+  return M[x , list_y]
+end
+
+#=
+  _mathnf(A::MatElem{fmpz}) -> MatElem{fmpz}, MatElem{fmpz}
+
+Given a rectangular matrix $A$ of dimension $nxm$. Compute the Hermite normal
+form $H$ of dimension $nxm$ and the unimodular transformation matrix $U$ such
+that $AU$ = $H$. The first n-rank(A) columns are zero and the rest are in 
+Gauß-form.
+=#
+function _mathnf(A::MatElem{fmpz})
+  H, U = hnf_with_transform(reverse_cols(transpose(A)))
+  H = reverse_rows(reverse_cols(transpose(H)))
+  U = reverse_cols(transpose(U))
+
+  return H, U
+end
+
+################################################################################
+#                           linear algebra
+################################################################################
+
+#=
+  _complete_to_basis(v::MatElem{fmpz}; redflag::Bool = false) -> MatElem{fmpz}
+
+Given a rectangular matrix $nxm$ with $n != m$, compute a unimodular matrix
+with the last column equal to the last column of $v$. If redflag = true,
+it LLL-reduce the $n-m$ first columns if $n > m$. 
+=#
+function _complete_to_basis(v::MatElem{fmpz}, redflag::Bool = false)
+  
+  n = nrows(v) 
+  m = ncols(v) 
+
+  if n == m 
+    return identity_matrix(v)
+  end
+
+  U = inv(transpose(_mathnf(transpose(v))[2]))
+
+  if n == 1 || redflag == false || m < n
```

Comment:
```suggestion
  if n == 1 || redflag == false || m > n
```
If $m<n$, according to your documentation, you want to reduce the $n-m$ first columns, so here you have to exit if you have the converse $m>n$